enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/replication/BenchmarkProcedure.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.causalclustering.core.replication;
+
+import java.io.IOException;
+import java.util.concurrent.Future;
+import java.util.stream.Stream;
+
+import org.neo4j.causalclustering.core.state.machines.dummy.DummyRequest;
+import org.neo4j.kernel.api.exceptions.InvalidArgumentsException;
+import org.neo4j.logging.Log;
+import org.neo4j.procedure.Context;
+import org.neo4j.procedure.Description;
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.Procedure;
+
+import static java.lang.Math.toIntExact;
+import static org.neo4j.procedure.Mode.DBMS;
+
+@SuppressWarnings( "unused" )
+public class BenchmarkProcedure
+{
+    @Context
+    public Replicator replicator;
+
+    @Context
+    public Log log;
+
+    private static long startTime;
+    private static Worker[] workerArr;
```

Comment:
Not a big deal at all, but I dislike Java arrays for reasons of syntax/API and `ArrayStoreException`. It is possible to create an `ArrayList` with an initial capacity.